Ratwood Pottery & Glass-Working Port

Stutternov opened the pull request:
## About The Pull Request

Right! This PR is **NEARLY** finished now!

### What Does This PR Add?

**POTTERY AND GLASS MAKING!** Yep, finally, passive-crafting stuff instead of wo-balance-jak fucking PRs.

Clay items can be seen here!
![image](https://github.com/user-attachments/assets/06e48ee5-06c6-403f-bf5e-dcf35c272513)

You can now make the following items out of naturally obtainable **clay**:
- Clay cups
- Clay bottles
- Clay vases
- FANCY clay vases (!!!)
- Clay statues

You can customize the colors on these using a dye bin!!!!

Now also windows take **glass** to make rather than *ash*! This ones a bit more bare-bones, as it's possibly good for future content, but here's a small list:
- Windows now take glass to make 
- Glass bottles can be made via glass-blowing
- Glass statues, which are valuable, can be made via glass-blowing
![image](https://github.com/user-attachments/assets/9620b1c1-9dfd-4267-8958-3bcc38f0ac8e)

Also - added **bricks** that are makeable using clay. (Is this how brickmaking works? No. But we're simplifying for fun.)

Currently though bricks can only make the prior unobtainable brick-floor type.

### Who Gets the New Ceramics Skill?

For now, JUST artificers and towner-potters. 

Artificers only get it just because we need someone to be able to do repairs, which you now need glass to make windows, so they need some skill in it to be fair. Potters will still exceed them.

Also may make artificers good backups for the steward if they need some valuable stuff to sell in a bind.

### Is This All?

For now, yes. **HOWEVER** - I do ask any spriters who are interested in helping to reach out. I have more stuff I want but, honestly, I cannot sprite for jack-shit. My future plans, if able:
- ~~Brick walls~~ - Done!
- ~~Brick windows~~ - Done!
- Glass art works (fancy bottles, stained glass types maybe, decorations, etc) - Kinda done, want more but did small bit.
- Clay kettle (raw and cooked, have an idea for a version too like an 'assassins pot' kinda deal)
- Various craftable structure-statues from clay.

### Cool!!! Did you make this???

I did some of the sprites and additions but no, SunriseOverYourHead did 90% of this entire PR with their code from Ratwood. You can find the PR below, it is REALLY good content. Props go to them!

 https://github.com/Rotwood-Vale/Ratwood-Keep/pull/1053

## Testing Evidence

(Soon)

## Why It's Good For The Game

Tl;dr section here.

Simply put - it's another crafting skill, but intent is to use this as a base to add stuff slowly with time since it's a really good idea (imo) to focus on building material aspects of the game and provide a few ways to interact with the economy without rare materials and mining.

Bricks and glass will make crafting projects more customizable and fit with the setting, plus encourage working together. While pottery and clay are simple and easy to do and have a reward of mildly valuable stuff with customization for looks as well as function.

SunriseOverYourHead commented:
Hi, I'm the guy in question.

You might want to check the .dmi s and replace the clay-related sprites with your own. When I made this PR had done only very basic coder-sprites that frankly looked rather poor and didn't fit the style of roguetown-like servers. (It's also why I didn't bother PRing it again after it was closed.) Hopefully you'll be able to work around that.

As for glass blowing, the plan was to have it be used to make expensive vases and pottery that could be sold as treasure instead of only using it to repair windows. Clay would be your bread-and-butter material for utility items while glass would be used for expensive and fancy things. Once again, I didn't get around to that because sadly painting skill doesn't translate to pixel art skill (And I'd rather not do something at all than do something poorly), but this might help you understand why I coded it in the way that I did.

Enjoy!

Stutternov commented on code/modules/roguetown/roguecrafting/turfs.dm:
```diff
@@ -305,3 +305,77 @@
 	return ..()
+
+// Normal, non-openable window
+/datum/crafting_recipe/roguetown/turfs/roguewindow
```
Will have to see in testing; I saw we have craftable windows which seeem to work fine but I'll double-check this one.

Stutternov commented on code/modules/roguetown/roguejobs/ceramicist/clay.dm:
```diff
@@ -0,0 +1,71 @@
+			name = "burnt [name]"
+			desc = "It's dark and charred, but still functional."
+		if(burning > (burntime * 2)) // Way too long. Incinerate.
+			burn() // Burn is a method of obj/items that Qdels something and replaces it with ash.
```
I think it's good to keep most of this given people will look at stuff going: "WHAT DOES THIS MEEEEAN!?"

Stutternov commented:
Oh-Kay, I did the brick wall code and sprites. Finally done. Just gotta test it again and upload images; it seemed fine in my prior test but I didn't have statues at the time so I'll have to run a test again with them and the brick walls.

Disheveled-arch commented:
I feel, as someone who occasionally does building, that the addition of glass as a crafting component for 'wooden windows' will curtail a fair portion of folks desire to build.  Construction projects already require pre-planning and a fair bit of your round dedicated to resource gathering and construction, often 2-2.5 hours of work for the bigger projects.  

I think glass is great for statues and bottles and the like, but the window is one of the few items that make a building feel like a complete building without having murder holes all over the place.   Adding this extra hurdle to construction, imo, pushes the prospect of building nice buildings into the 'Just not worth my time' category.

Perhaps if you introduced the ability to craft the openable windows, they could use glass due to their higher complexity and functionality, but It feels restrictive on the basic 'Wooden Window'.

Stutternov commented:
Added some new sprites for the ceramics curtousy of Witty, sprites by Cre.
![image](https://github.com/user-attachments/assets/6c9552b7-bc7f-43bd-b2cc-9fcbba9d70a6)

Replaced the cup with a flask.

codyman765 commented:
So, played around with this a bit RoundID:Sherwood1807-13.05.36

Its... eeesh.

Me and a friend spent about 20 minutes trying to dig up clay only to get nothing, from what I understand it should drop from digging in normal, non muddy dirt?

Ended up using the few bits that came with the towner role after making a wheel, made a bottle and a fancy vase. After looking through both this PR and the OG one, figured they must go into a oven, both items gave me the "Something smells good" notice then instantly turned into ash. Wasn't able to get anymore clay to test more. But if its not made for the oven- i'm not sure what else, the comments in the code refer a kiln but I don't see one added by this PR, and the port says they didn't go with making a kiln due to bloat worry.

Overall If you DO get clay from digging, and I wasn't doing it wrong, the drop rate really needs to be buffed. but im just hoping I was doing it wrong. As for if they are used in a oven.. Im still not sure.

codyman765 commented:
Round ID :Spike1828-08.53.14

Figured id give it another go since I saw the increased clay chance. Dug for ten mins straight the shovel given from the potter role. Got 42 stones and zero clay. It IS dug up the same way you get stones correct? Since otherwise im not seeing any at all. Use the role started clay to make a bottle and statue and tossed em into a oven. Both turn into ash instantly after getting the "Something smells good" message.

Stutternov commented:
So, did a fix to the cooking. It was working after a simple change, so assuming maybe that fixed it. Only issue I'm having: Cook sprites don't have a sprite, and struggling to figure out why. (It has an in-hand inventory sprite, just no on-floor/on table one.)
![image](https://github.com/user-attachments/assets/56c634cf-cecb-4985-b811-7a39f6d35472)

As for struggling to dig up clay, I've actually not had the issue. You just have to dig in non-muddy tiles, it actually gives you clay more than it gives you rocks so far.
![image](https://github.com/user-attachments/assets/068b3511-4165-420a-8a0d-e0d2ecf5a05e)

codyman765 commented:
> So, did a fix to the cooking. It was working after a simple change, so assuming maybe that fixed it. Only issue I'm having: Cook sprites don't have a sprite, and struggling to figure out why. (It has an in-hand inventory sprite, just no on-floor/on table one.) ![image](https://private-user-images.githubusercontent.com/47883419/436717350-56c634cf-cecb-4985-b811-7a39f6d35472.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDU0NTY1MzYsIm5iZiI6MTc0NTQ1NjIzNiwicGF0aCI6Ii80Nzg4MzQxOS80MzY3MTczNTAtNTZjNjM0Y2YtY2VjYi00OTg1LWI4MTEtN2EzOWY2ZDM1NDcyLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA0MjQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNDI0VDAwNTcxNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWQ0NjJiYjI0NDFmNjNjZDE0NjQ3M2FkOTQ5NWExODQ0ZDlmNDRlMTM1ZWYyOTBhNGFlZDIyNWYwZDcwNzVlMTkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.oIgRTzzCpJMaJ_Sszl5lq7HKMa_e3iBg0qXV8M-fX1E)
> 
> As for struggling to dig up clay, I've actually not had the issue. I'd need a replication and example how many rocks gathered vs how much clay and on what tile-type. You just have to dig in non-muddy tiles, it actually gives you clay more than it gives you rocks so far. ![image](https://private-user-images.githubusercontent.com/47883419/436717514-068b3511-4165-420a-8a0d-e0d2ecf5a05e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDU0NTY1MzYsIm5iZiI6MTc0NTQ1NjIzNiwicGF0aCI6Ii80Nzg4MzQxOS80MzY3MTc1MTQtMDY4YjM1MTEtNDE2NS00MjBhLThhMGQtZTBkMmVjZjVhMDVlLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA0MjQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNDI0VDAwNTcxNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTgxMjRiYjFkM2VjMjU3Y2RiOWMxMzU5YzBkNGNkNDI2NzhjZDJhOTE2YzA4MzA0YmI1ZDYwNWM1Y2Y3NGY5ZDYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.CEUNFFiMCpUaB40oTWWWu4cnlM8bZHrsEqHlXDrF8gg)
> 
> If I can get a fix for the former, this PR is basically good to go.

I can make another test when I get home from work tonight, but was digging in normal dirt, near the south River. Non muddy tile. Would dig a small hole, the first step of digging. Then fill it in with the dirt just dug up and repeated until a stone popped up, then would remove the stone and resume digging. This was the pattern that ended up giving me 42 stone to zero clay during my test. I've also seen other people say when I asked about it in roundstart OOC that they also are unable to find clay.

I'm honestly floored you were able to get clay and I wonder what the difference done was. I feel like there should have been at least one clay after 42 stones.

Stutternov commented:
> I'm honestly floored you were able to get clay and I wonder what the difference done was. I feel like there should have been at least one clay after 42 stones.

If you can, that was my thiiird shovel? First one resulted in stone, second nothing, third clay. And afterwards I kept trying just to ensure it wasn't a duplicate and got 3 clay and 2 stones total after a few more shovels.

I'm unsure if I just had amazing luck or if the spawn rate is too low and my test was just a lucky test.

codyman765 commented:
> > I'm honestly floored you were able to get clay and I wonder what the difference done was. I feel like there should have been at least one clay after 42 stones.
> 
> If you can, that was my thiiird shovel? First one resulted in stone, second nothing, third clay. And afterwards I kept trying just to ensure it wasn't a duplicate and got 3 clay and 2 stones total after a few more shovels.
> 
> I'm unsure if I just had amazing luck or if the spawn rate is too low and my test was just a lucky test. I did up it from 30% to 40% IF no stone is found since the initial comment that the spawn rate was too low.

Out of curiosity, are you leaving the stone on the tile you are digging on, or moving it off? I "think" that another stone won't spawn if there is already a stone there. Which, I guess might force consistent clay if I'm correct now that I'm thinking about it.

Stutternov commented:
> Out of curiosity, are you leaving the stone on the tile you are digging on, or moving it off? I "think" that another stone won't spawn if there is already a stone there. Which, I guess might force consistent clay if I'm correct now that I'm thinking about it.

I did, I left it on the tile after the second stone. Though I'm unsure if they will stop generation. Worth a shot in testing.

codyman765 commented:
> > Out of curiosity, are you leaving the stone on the tile you are digging on, or moving it off? I "think" that another stone won't spawn if there is already a stone there. Which, I guess might force consistent clay if I'm correct now that I'm thinking about it.
> 
> I did, I left it on the tile after the second stone. Though I'm unsure if they will stop generation. Worth a shot in testing.

Okay! I did some testing, edited some code and ran it on a personal server to test and it seems to work- though I didn't have the clay to test.

So it seems currently the way the digging code is set up, is on IF statements. and by checking if there is already an item on the tile before spawning it. Such as it checks if there is a stone, if there is it returns to end, preventing another stone from being dug up until its removed. Due to the way the code is set up currently, it looks if there is a stone>If stone do not spawn stone, move onto clay. It only goes to do the clay stuff **IF** a stone is present. Otherwise it returns and ends.

What I ended up doing on my small test was removing the if(!(locate lines, in both mud and the normal one. meaning it was no longer checking if an item was there before spawning, it moves straight into the dice roll to see if the item spawns. Doing this also made it so you didn't need to clear the tile before you were able to dig anything up. This worked for both stones, and worms/leeches

Im guessing it may have been set up this way to prevent people from digging up alot of stuff quickly? But I am unsure

Might be able to restructure the code so its rolling to spawn both stone and clay at once, rather then relying on it already finding a stone.

Otherwise you _could_ remove both of the if(!(locate lines, fix the formatting left behind and lower the % chances a bit to help with the quicker digging now. Seemed to work fine on my limited testing on a private server.

Personally it would be nice to not have to clear the tile before getting a new stone, speeding up stone gathering/projects. But I'm not sure why it was set up this way in the first place. If this doesn't make sense- sorry just got home from 10hr shift and tested this all.

TDLR- Clay only spawns if it sees stone in the spot. Seems mud follows the same rules for worms/leaches. Needs one present before it moves onto the other type.
